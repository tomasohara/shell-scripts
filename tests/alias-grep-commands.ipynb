{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9a234961",
   "metadata": {},
   "outputs": [],
   "source": [
    "## SETUP (OPTIONAL - USE IF REQUIRED)\n",
    "## Bracketed Paste is disabled to prevent characters after output\n",
    "## Example: \n",
    "## $ echo 'Hii'\n",
    "## | Hi?2004l\n",
    "# bind 'set enable-bracketed-paste off'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "36ea45cf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "## TEST:  Make sure simple prompt used (e.g., no escapes that might contaminate output)\n",
    "## PS1=\"$ \"\n",
    "## TODO: PS1=\"> \"\n",
    "\n",
    "## NOTE: The Jupyter bash kernel requires that PS1 not be modified as they customize it.\n",
    "# echo \"TODO:\"\n",
    "# echo \"  echo \\\"\\$PS1\\\" | command grep --count --perl-regexp '[\\x00-\\x1f]'\"\n",
    "\n",
    "actual=${PS1: -1}\n",
    "[ \"$actual\" = \">\" ]; echo $?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "118f4d68",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "# Setting a temp directory for tests\n",
    "[ \"$DEBUG_LEVEL\" -lt 4 ]; echo \"$?\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "58f47ac9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "\n",
    "## Aliases for censoring username and/or numbers\n",
    "alias testnum=\"sed -r \"s/[0-9]/N/g\"\" \n",
    "alias testuser=\"sed -r \"s/\"$USER\"+/userxf333/g\"\"\n",
    "alias testuserpath=\"sed 's|/home/[^/]\\+/|USER_PATH/|'\"\n",
    "alias linebr=\"printf '%132s\\n' ' ' | tr ' ' '-'\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "482c3dcb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed '/tmp/test-grep-commands/alias-grep-commands' -> '/tmp/test-grep-commands/alias-grep-commands.16Apr24.9'\n"
     ]
    }
   ],
   "source": [
    "# Global setup\n",
    "\n",
    "## Setting a temp directory for tests\n",
    "## NOTE: If debugging uses $TMP\n",
    "## Use of rm command is heavily discouraged, use move to $trash_dir instead\n",
    "\n",
    "TMP=${TMP:-/tmp}\n",
    "if [ \"$DEBUG_LEVEL\" -lt 4 ]; then TMP=/tmp/test-grep-commands; fi\n",
    "\n",
    "## OLD: Used the concept of trash-dir, instead use unique temp_dirs\n",
    "# temp_dir=$TMP/test-7371\n",
    "# trash_dir=$TMP/\"_temp-trash-$(date \"+%Y%m%d%H%M%S\")\"\n",
    "# command mkdir -p $temp_dir\n",
    "# command mkdir -p $trash_dir\n",
    "# command cd $temp_dir\n",
    "\n",
    "temp_dir=$TMP/alias-grep-commands\n",
    "rename-with-file-date \"$temp_dir\"\n",
    "command mkdir -p \"$temp_dir\"\n",
    "command cd \"$temp_dir\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ab6448b5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "## Count aliases w/ -dir in name\n",
    "\n",
    "## Example:\n",
    "##   $ alias | egrep '\\-dir\\b' | wc -l\n",
    "##   29\n",
    "\n",
    "dir_aliases=$(alias | egrep '\\-dir\\b' | wc -l)\n",
    "[ \"$dir_aliases\" -gt 20 ]; echo \"$?\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4eb0b066",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "## Count functions w/ -dir in name\n",
    "\n",
    "## Example:\n",
    "## $ typeset -f | egrep '^\\w+' | wc -l\n",
    "## | 474\n",
    "\n",
    "dir_functions=$(typeset -f | egrep '\\-dir\\b' | wc -l)\n",
    "[ \"$dir_functions\" -gt 35 ]; echo \"$?\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "993a90d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD\n",
    "# source $BIN_DIR/tomohara-aliases.bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "37c6221e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "grep (GNU grep) 3.7\n"
     ]
    }
   ],
   "source": [
    "grep_ver=$(grep -V | head -n 1 | tail )\n",
    "echo $grep_ver"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "135998e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "\n",
    "## SETTING UP FOR GREP TEST\n",
    "## BAD: rm -rf ./* > /dev/null\n",
    "# Not recommended as not contents in directory: mv -f ./* $trash_dir\n",
    "\n",
    "# THE WORD TO BE TESTED - sensitive\n",
    "command mkdir testdir1 testdir2\n",
    "echo \"As grep commands are case sensitive, one of the most useful operators for grep searches as they are sensitive is -i. Instead of printing lowercase results only, the terminal displays both uppercase and lowercase results. The output includes lines with mixed case entries.\n",
    "\" > testgrep1\n",
    "echo \"sensitive sensitive sensitive Sensitive SENSITIVE lorem ipsum\" > testgrep2\n",
    "echo \"no mentions here\" > testgrep3\n",
    "echo \"Passwords are generally case sensitive\" > testgrep4\n",
    "printf \"Non ASCII: Ñ\\nNext\\nNon ASCII: §\" > testgrep5\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f906642e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[35m\u001b[Ktestgrep1\u001b[m\u001b[K\u001b[36m\u001b[K:\u001b[m\u001b[KAs grep commands are case \u001b[01;31m\u001b[Ksensitive\u001b[m\u001b[K, one of the most useful operators for grep searches as they are \u001b[01;31m\u001b[Ksensitive\u001b[m\u001b[K is -i. Instead of printing lowercase results only, the terminal displays both uppercase and lowercase results. The output includes lines with mixed case entries.\n",
      "\u001b[35m\u001b[Ktestgrep2\u001b[m\u001b[K\u001b[36m\u001b[K:\u001b[m\u001b[K\u001b[01;31m\u001b[Ksensitive\u001b[m\u001b[K \u001b[01;31m\u001b[Ksensitive\u001b[m\u001b[K \u001b[01;31m\u001b[Ksensitive\u001b[m\u001b[K Sensitive SENSITIVE lorem ipsum\n",
      "\u001b[35m\u001b[Ktestgrep4\u001b[m\u001b[K\u001b[36m\u001b[K:\u001b[m\u001b[KPasswords are generally case \u001b[01;31m\u001b[Ksensitive\u001b[m\u001b[K\n"
     ]
    }
   ],
   "source": [
    "## TEST-2: TESTING GREP ALIASES\n",
    "\n",
    "# 0) grep \n",
    "grep sensitive testgrep1 testgrep2 testgrep3 testgrep4\n",
    "\n",
    "## OLD\n",
    "# linebr\n",
    "# # A) gu (grep-unique i-)\n",
    "# gu sensitive testgrep1 testgrep2 testgrep3 testgrep4\n",
    "# linebr\n",
    "# # B) gu- (grep-unique)\n",
    "# gu- sensitive testgrep1 testgrep2 testgrep3 testgrep4\n",
    "# linebr\n",
    "# # C) grepl- (grep-to-less)\n",
    "# grepl- \"sensitive\" testgrep1 testgrep2 testgrep3 testgrep4 | wc -l\n",
    "# # NOTE: wc -l is pipelined as command goes to a loop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "b209fa0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "testgrep4:1\n",
      "testgrep2:1\n",
      "testgrep1:1\n"
     ]
    }
   ],
   "source": [
    "# A) gu (grep-unique i-)\n",
    "gu sensitive testgrep1 testgrep2 testgrep3 testgrep4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "66375e3e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "testgrep4:1\n",
      "testgrep2:1\n",
      "testgrep1:1\n"
     ]
    }
   ],
   "source": [
    "# B) gu- (grep-unique)\n",
    "gu- sensitive testgrep1 testgrep2 testgrep3 testgrep4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "4a61dbfa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\n"
     ]
    }
   ],
   "source": [
    "# C) grepl- (grep-to-less)\n",
    "grepl- \"sensitive\" testgrep1 testgrep2 testgrep3 testgrep4 | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "690311c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Halts test\n",
    "# show-line-context testgrep3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c1709353",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1:sensitive sensitive sensitive Sensitive SENSITIVE lorem ipsum\n"
     ]
    }
   ],
   "source": [
    "## OLD\n",
    "cat testgrep2 | gr-less lorem"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4221570d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Non ASCII: Ñ\n",
      "Non ASCII: §\n"
     ]
    }
   ],
   "source": [
    "cat testgrep5 | grep-nonascii"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "76128375",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Need help for: fs-, findgrep, fgr, fgr-py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e51dcd01",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-r--r-- N userxfNNN userxfNNN N  NN NN:NN ./ls-alR.list\n",
      "-rw-r--r-- N userxfNNN userxfNNN NN  NN NN:NN ./ls-alR.list.log\n",
      "-rw-r--r-- N userxfNNN userxfNNN N  NN NN:NN ./ls-aR.list\n",
      "-rw-r--r-- N userxfNNN userxfNNN NN  NN NN:NN ./ls-aR.list.log\n"
     ]
    }
   ],
   "source": [
    "prepare-find-files-here | testuser | testnum | awk '!($6=\"\")'\n",
    "# prepare-find-files-here | sed 's|\\([^ ]\\+\\) \\([^ ]\\+\\) \\([^ ]\\+\\) \\([^ ]\\+\\) \\([^ ]\\+\\) \\([^ ]\\+\\) \\([^ ]\\+\\) \\([^ ]\\+\\) \\(.*\\)|\\1 \\6 \\7 \\8|'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "584ac3a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mbackup\u001b[0m       ls-alR.list.log  ls-aR.list.log  \u001b[01;34mtestdir2\u001b[0m   testgrep2  testgrep4\n",
      "ls-alR.list  ls-aR.list       \u001b[01;34mtestdir1\u001b[0m        testgrep1  testgrep3  testgrep5\n"
     ]
    }
   ],
   "source": [
    "command mkdir backup/\n",
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "4da38cda",
   "metadata": {},
   "outputs": [],
   "source": [
    "find-files \"ls*\" | testuser | testnum | awk '!($6=\"\")'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "f7c048b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Need Help (Non-ending process)\n",
    "# $ find-files- \"testgrep4\" | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "1572700d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # # ISSUE\n",
    "# # $ make-file-listing\n",
    "# | cp: cannot stat 'ls-aR.list': No such file or directory\n",
    "# | chmod: cannot access './backup/ls-aR.list': No such file or directory\n",
    "## BAD: command mv -f * $trash_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "dc3f2836",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "END\n"
     ]
    }
   ],
   "source": [
    "echo \"END\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
