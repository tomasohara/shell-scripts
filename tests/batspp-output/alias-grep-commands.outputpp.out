1..17
not ok 1 test-1
# (in test file batspp-output/alias-grep-commands.outputpp, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== echo $PS1
#  ==========
#
# ========= [PEXP\[\]ECT_PROMPT> =========
# [PEXP\[\]ECT_PROMPT>
# ============================
not ok 2 test-3
# (in test file batspp-output/alias-grep-commands.outputpp, line 59)
#   `[ "$(test-3-actual)" == "$(test-3-expected)" ]' failed
# ========== unalias -a
#  ==========
# ========= $ alias | wc -l\n$ for f in ; do unset -f ; done\n$ typeset -f | egrep '^\w+' | wc -l\n0\n0 =========
# $ alias | wc -l
# $ for f in bats_abort_timeout_countdown ()
# bats_add_debug_exclude_path ()
# bats_all_in ()
# bats_any_in ()
# bats_append_arrays_as_args ()
# bats_approx_realpath ()
# bats_binary_search ()
# bats_capture_stack_trace ()
# bats_check_status_from_trap ()
# bats_check_valid_version ()
# bats_create_test_tmpdirs ()
# bats_debug_trap ()
# bats_emit_trace ()
# bats_error_trap ()
# bats_evaluate_preprocessed_source ()
# bats_execute ()
# bats_exit_trap ()
# bats_extract_line ()
# bats_format_file_line_reference ()
# bats_format_file_line_reference_colon ()
# bats_format_file_line_reference_comma_line ()
# bats_format_file_line_reference_uri ()
# bats_frame_filename ()
# bats_frame_function ()
# bats_frame_lineno ()
# bats_generate_warning ()
# bats_get_child_processes_of ()
# bats_get_failure_stack_trace ()
# bats_internal_load ()
# bats_interrupt_trap ()
# bats_interrupt_trap_in_run ()
# bats_kill_childprocesses_of ()
# bats_load_library ()
# bats_load_library_safe ()
# bats_load_safe ()
# bats_merge_stdout_and_stderr ()
# bats_normalize_windows_dir_path ()
# bats_perform_test ()
# bats_prefix_lines_for_tap_output ()
# bats_preprocess_source ()
# bats_print_failed_command ()
# bats_print_stack_trace ()
# bats_quote_code ()
# bats_readlinkf ()
# declare -fx bats_readlinkf
# bats_redirect_stderr_into_file ()
# bats_replace_filename ()
# bats_require_minimum_version ()
# bats_separate_lines ()
# bats_setup_tracing ()
# bats_should_retry_test ()
# bats_sort ()
# bats_start_timeout_countdown ()
# bats_strip_string ()
# bats_teardown_trap ()
# bats_test_begin ()
# bats_test_function ()
# bats_timeout_trap ()
# bats_trim ()
# bats_trim_filename ()
# bats_version_lt ()
# bats_warn_minimum_guaranteed_version ()
# find_in_bats_lib_path ()
# get_mills_since_epoch ()
# load ()
# run ()
# setup ()
# skip ()
# teardown ()
# test-1-actual ()
# test-1-expected ()
# END_EXPECTED
# test-10-actual ()
# test-10-expected ()
# grep (GNU grep) N.N
# Copyright (C) NNNN Free Software Foundation, Inc.
# License GPLvN+: GNU GPL version N or later <https://gnu.org/licenses/gpl.html>.
# This is free software: you are free to change and redistribute it.
# There is NO WARRANTY, to the extent permitted by law.
# END_EXPECTED
# test-11-actual ()
# test-11-expected ()
# END_EXPECTED
# test-12-actual ()
# test-12-expected ()
# testgrep1:As grep commands are case sensitive, one of the most useful operators for grep searches as they are sensitive is -i. Instead of printing lowercase results only, the terminal displays both uppercase and lowercase results. The output includes lines with mixed case entries.
# testgrep2:sensitive sensitive sensitive Sensitive SENSITIVE lorem ipsum
# testgrep4:Passwords are generally case sensitive
# testgrep4:1
# testgrep2:1
# testgrep1:1
# testgrep4:1
# testgrep2:1
# testgrep1:1
# 3
# END_EXPECTED
# test-13-actual ()
# test-13-expected ()
# END_EXPECTED
# test-14-actual ()
# test-14-expected ()
# Non ASCII: ร
# Non ASCII: ยง
# END_EXPECTED
# test-15-actual ()
# test-15-expected ()
# END_EXPECTED
# test-16-actual ()
# test-16-expected ()
# backup	     ls-alR.list.log  ls-R.list.log  testdir2	testgrep2  testgrep4
# ls-alR.list  ls-R.list	      testdir1	     testgrep1	testgrep3  testgrep5
# END_EXPECTED
# test-17-actual ()
# test-17-expected ()
# total NN
# total N
# total N
# END_EXPECTED
# test-18-actual ()
# test-18-expected ()
# END_EXPECTED
# test-3-actual ()
# test-3-expected ()
# 0
# 0
# END_EXPECTED
# test-4-actual ()
# test-4-expected ()
# 0
# END_EXPECTED
# test-5-actual ()
# test-5-expected ()
# END_EXPECTED
# test-6-actual ()
# test-6-expected ()
# END_EXPECTED
# test-7-actual ()
# test-7-expected ()
# 3
# END_EXPECTED
# test-8-actual ()
# test-8-expected ()
# 0
# END_EXPECTED
# test-9-actual ()
# test-9-expected ()
# END_EXPECTED
# test_test-2d1 ()
# test_test-2d10 ()
# grep (GNU grep) N.N
# Copyright (C) NNNN Free Software Foundation, Inc.
# License GPLvN+: GNU GPL version N or later <https://gnu.org/licenses/gpl.html>.
# This is free software: you are free to change and redistribute it.
# There is NO WARRANTY, to the extent permitted by law.' "=========";
# test_test-2d11 ()
# test_test-2d12 ()
# testgrep1:As grep commands are case sensitive, one of the most useful operators for grep searches as they are sensitive is -i. Instead of printing lowercase results only, the terminal displays both uppercase and lowercase results. The output includes lines with mixed case entries.
# testgrep2:sensitive sensitive sensitive Sensitive SENSITIVE lorem ipsum
# testgrep4:Passwords are generally case sensitive
# testgrep4:1
# testgrep2:1
# testgrep1:1
# testgrep4:1
# testgrep2:1
# testgrep1:1
# 3' "=========";
# test_test-2d13 ()
# test_test-2d14 ()
# Non ASCII: ยง' "=========";
# test_test-2d15 ()
# test_test-2d16 ()
# backup	     ls-alR.list.log  ls-R.list.log  testdir2	testgrep2  testgrep4
# ls-alR.list  ls-R.list	      testdir1	     testgrep1	testgrep3  testgrep5' "=========";
# test_test-2d17 ()
# total NN
# total N
# total N' "=========";
# test_test-2d18 ()
# test_test-2d3 ()
# test_test-2d4 ()
# 0' "=========";
# test_test-2d5 ()
# test_test-2d6 ()
# test_test-2d7 ()
# test_test-2d8 ()
# test_test-2d9 () ; do unset -f ; done
# $ typeset -f | egrep '^\w+' | wc -l
# 0
# 0
# ============================
not ok 3 test-4
# (in test file batspp-output/alias-grep-commands.outputpp, line 97)
#   `[ "$(test-4-actual)" == "$(test-4-expected)" ]' failed
# ========== TMP=/tmp/test-grep
#  ==========
# ========= ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=$PWD/..
# ## You will need to run jupyter from that directory.
# $ alias | wc -l
# 0 =========
# ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=/tmp/batspp-74694/test-4/..
# ## You will need to run jupyter from that directory.
# $ alias | wc -l
# 0
# ============================
not ok 4 test-5
# (in test file batspp-output/alias-grep-commands.outputpp, line 135)
#   `[ "$(test-5-actual)" == "$(test-5-expected)" ]' failed
# ========== alias testnum="sed -r "s/[0-9]/N/g""
#  ==========
# ========= $ alias testuser="sed -r "s/"$USER"+/userxf333/g"" =========
# $ alias testuser="sed -r "s/"aveey"+/userxf333/g""
# ============================
not ok 5 test-6
# (in test file batspp-output/alias-grep-commands.outputpp, line 169)
#   `[ "$(test-6-actual)" == "$(test-6-expected)" ]' failed
# ========== temp_dir=$TMP/test-7371
#  ==========
# ========= $ mkdir -p "$temp_dir"
# # TODO: /bin/rm -rvf "$temp_dir"
# $ cd "$temp_dir"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -"
# /tmp/test-grep/test-7371 =========
# $ mkdir -p "/test-7371"
# # TODO: /bin/rm -rvf "/test-7371"
# $ cd "/test-7371"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "80" '' | tr ' ' -"
# /tmp/test-grep/test-7371
# ============================
not ok 6 test-7
# (in test file batspp-output/alias-grep-commands.outputpp, line 209)
#   `[ "$(test-7-actual)" == "$(test-7-expected)" ]' failed
# ========== alias | wc -l
#  ==========
# 0
# ========= 3 =========
# 3
# ============================
not ok 7 test-8
# (in test file batspp-output/alias-grep-commands.outputpp, line 243)
#   `[ "$(test-8-actual)" == "$(test-8-expected)" ]' failed
# ========== typeset -f | egrep '^\w+' | wc -l\n ==========
# 186
# ========= 0 =========
# 0
# ============================
not ok 8 test-9
# (from function `test-9-actual' in file batspp-output/alias-grep-commands.outputpp, line 284,
#  in test file batspp-output/alias-grep-commands.outputpp, line 271)
#   `test-9-actual ' failed
# ========== source $BIN_DIR/tomohara-aliases.bash
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 284: /tomohara-aliases.bash: No such file or directory
not ok 9 test-10
# (in test file batspp-output/alias-grep-commands.outputpp, line 311)
#   `[ "$(test-10-actual)" == "$(test-10-expected)" ]' failed
# ========== # $ grep -V
#  ==========
# ========= # grep (GNU grep) 2.4.2
# #
# # Copyright 1988, 1992-1999, 2000 Free Software Foundation, Inc.
# # ...
# $ grep -V | testnum
# grep (GNU grep) N.N
# Copyright (C) NNNN Free Software Foundation, Inc.
# License GPLvN+: GNU GPL version N or later <https://gnu.org/licenses/gpl.html>.
# This is free software: you are free to change and redistribute it.
# There is NO WARRANTY, to the extent permitted by law. =========
# # grep (GNU grep) 2.4.2
# #
# # Copyright 1988, 1992-1999, 2000 Free Software Foundation, Inc.
# # ...
# $ grep -V | testnum
# grep (GNU grep) N.N
# Copyright (C) NNNN Free Software Foundation, Inc.
# License GPLvN+: GNU GPL version N or later <https://gnu.org/licenses/gpl.html>.
# This is free software: you are free to change and redistribute it.
# There is NO WARRANTY, to the extent permitted by law.
# ============================
not ok 10 test-11
# (in test file batspp-output/alias-grep-commands.outputpp, line 353)
#   `[ "$(test-11-actual)" == "$(test-11-expected)" ]' failed
# ========== rm -rf ./* > /dev/null
#  ==========
# ========= # THE WORD TO BE TESTED - sensitive
# $ mkdir testdir1 testdir2
# $ echo "As grep commands are case sensitive, one of the most useful operators for grep searches as they are sensitive is -i. Instead of printing lowercase results only, the terminal displays both uppercase and lowercase results. The output includes lines with mixed case entries.
# $ " > testgrep1
# $ echo "sensitive sensitive sensitive Sensitive SENSITIVE lorem ipsum" > testgrep2
# $ echo "no mentions here" > testgrep3
# $ echo "Passwords are generally case sensitive" > testgrep4
# $ printf "Non ASCII: ร\nNext\nNon ASCII: ยง" > testgrep5 =========
# # THE WORD TO BE TESTED - sensitive
# $ mkdir testdir1 testdir2
# $ echo "As grep commands are case sensitive, one of the most useful operators for grep searches as they are sensitive is -i. Instead of printing lowercase results only, the terminal displays both uppercase and lowercase results. The output includes lines with mixed case entries.
# $ " > testgrep1
# $ echo "sensitive sensitive sensitive Sensitive SENSITIVE lorem ipsum" > testgrep2
# $ echo "no mentions here" > testgrep3
# $ echo "Passwords are generally case sensitive" > testgrep4
# $ printf "Non ASCII: ร\nNext\nNon ASCII: ยง" > testgrep5
# ============================
not ok 11 test-12
# (from function `test-12-actual' in file batspp-output/alias-grep-commands.outputpp, line 401,
#  in test file batspp-output/alias-grep-commands.outputpp, line 388)
#   `test-12-actual ' failed with status 2
# ========== grep sensitive testgrep1 testgrep2 testgrep3 testgrep4
#  ==========
# grep: testgrep1: No such file or directory
# grep: testgrep2: No such file or directory
# grep: testgrep3: No such file or directory
# grep: testgrep4: No such file or directory
not ok 12 test-13
# (from function `test-13-actual' in file batspp-output/alias-grep-commands.outputpp, line 457,
#  in test file batspp-output/alias-grep-commands.outputpp, line 444)
#   `test-13-actual ' failed with status 127
# ========== cat testgrep2 | gr-less lorem
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 457: gr-less: command not found
# cat: testgrep2: No such file or directory
not ok 13 test-14
# (from function `test-14-actual' in file batspp-output/alias-grep-commands.outputpp, line 491,
#  in test file batspp-output/alias-grep-commands.outputpp, line 478)
#   `test-14-actual ' failed with status 127
# ========== cat testgrep5 | grep-nonascii
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 491: grep-nonascii: command not found
# cat: testgrep5: No such file or directory
not ok 14 test-15
# (in test file batspp-output/alias-grep-commands.outputpp, line 519)
#   `[ "$(test-15-actual)" == "$(test-15-expected)" ]' failed
# ========== prepare-find-files-here | testuser | testnum | awk '!($6="")'
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 526: prepare-find-files-here: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 526: testuser: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 526: testnum: command not found
# ========= -rw-rw-r-- N userxfNNN userxfNNN NNN  N NN:NN ls-alR.list
# -rw-rw-r-- N userxfNNN userxfNNN N  N NN:NN ls-alR.list.log
# -rw-rw-r-- N userxfNNN userxfNNN NNN  N NN:NN ls-R.list
# -rw-rw-r-- N userxfNNN userxfNNN N  N NN:NN ls-R.list.log =========
# -rw-rw-r-- N userxfNNN userxfNNN NNN  N NN:NN ls-alR.list
# -rw-rw-r-- N userxfNNN userxfNNN N  N NN:NN ls-alR.list.log
# -rw-rw-r-- N userxfNNN userxfNNN NNN  N NN:NN ls-R.list
# -rw-rw-r-- N userxfNNN userxfNNN N  N NN:NN ls-R.list.log
# ============================
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 526: prepare-find-files-here: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 526: testuser: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 526: testnum: command not found
not ok 15 test-16
# (in test file batspp-output/alias-grep-commands.outputpp, line 556)
#   `[ "$(test-16-actual)" == "$(test-16-expected)" ]' failed
# ========== mkdir backup/
#  ==========
# ========= $ ls
# backup	     ls-alR.list.log  ls-R.list.log  testdir2	testgrep2  testgrep4
# ls-alR.list  ls-R.list	      testdir1	     testgrep1	testgrep3  testgrep5 =========
# $ ls
# backup	     ls-alR.list.log  ls-R.list.log  testdir2	testgrep2  testgrep4
# ls-alR.list  ls-R.list	      testdir1	     testgrep1	testgrep3  testgrep5
# ============================
# mkdir: cannot create directory โbackup/โ: File exists
not ok 16 test-17
# (in test file batspp-output/alias-grep-commands.outputpp, line 592)
#   `[ "$(test-17-actual)" == "$(test-17-expected)" ]' failed
# ========== find-files "ls*" | testuser | testnum | awk '!($6="")'
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 599: find-files: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 599: testuser: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 599: testnum: command not found
# ========= .:
# total NN
# -rw-rw-r-- N userxfNNN userxfNNN N  N NN:NN ls-alR.list
# -rw-rw-r-- N userxfNNN userxfNNN N  N NN:NN ls-alR.list.log
# -rw-rw-r-- N userxfNNN userxfNNN NNN  N NN:NN ls-R.list
# -rw-rw-r-- N userxfNNN userxfNNN N  N NN:NN ls-R.list.log
# ./testdirN:
# total N
# ./testdirN:
# total N =========
# .:
# total NN
# -rw-rw-r-- N userxfNNN userxfNNN N  N NN:NN ls-alR.list
# -rw-rw-r-- N userxfNNN userxfNNN N  N NN:NN ls-alR.list.log
# -rw-rw-r-- N userxfNNN userxfNNN NNN  N NN:NN ls-R.list
# -rw-rw-r-- N userxfNNN userxfNNN N  N NN:NN ls-R.list.log
# ./testdirN:
# total N
# ./testdirN:
# total N
# ============================
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 599: find-files: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 599: testuser: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-grep-commands.outputpp: line 599: testnum: command not found
ok 17 test-18
